flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/factories/BooleanSchemaFactory.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.pulsar.common.schema.factories;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
+import org.apache.flink.connector.pulsar.common.schema.PulsarSchemaFactory;
+
+import org.apache.pulsar.client.api.Schema;
+import org.apache.pulsar.client.impl.schema.BooleanSchema;
+import org.apache.pulsar.common.schema.SchemaInfo;
+import org.apache.pulsar.common.schema.SchemaType;
+
+import java.util.Objects;
+
+/** The schema factory for pulsar's {@link BooleanSchema}. */
+public class BooleanSchemaFactory implements PulsarSchemaFactory<Boolean> {
```

Comment:
I think this and 90% of the other factories can be replaced with
```
public class PrimitiveSchemaFactory<T> implements PulsarSchemaFactory<T> {

    private final Schema<T> schema;
    private final TypeInformation<T> typeInformation;

    public PrimitiveSchemaFactory(Schema<T> schema, TypeInformation<T> typeInformation) {
        this.schema = schema;
        this.typeInformation = typeInformation;
    }

    @Override
    public SchemaType type() {
        return schema.getSchemaInfo().getType();
    }

    @Override
    public boolean supported(SchemaInfo info) {
        return info.getType() == type()
                && Objects.equals(schema.getSchemaInfo().getName(), info.getName());
    }

    @Override
    public Schema<T> createSchema(SchemaInfo info) {
        return schema;
    }

    @Override
    public TypeInformation<T> createTypeInfo(SchemaInfo info) {
        return typeInformation;
    }
}
```
We still need the interface for complex types (map, avro, json, protobuf).